### PR TITLE
[CMake][ROOT-9637] Adopt c++14 flag in root-config

### PR DESCRIPTION
**Pull request by dpiparo**
unless we are not on osx and gcc the gcc major is 4.

### CODE REVIEW

File: config/root-config.in [hunk 1]
```diff
@@ -146,8 +146,8 @@ fi
 ### machine dependent settings ###
 
 case "$cxxversion" in
-  cxx17) cxxversionflag="-std=c++1z " ;;
-  cxx14) cxxversionflag="-std=c++1y " ;;
+  cxx17) cxxversionflag="-std=${cxxversion} " ;;
+  cxx14) cxxversionflag="-std=${cxxversion} " ;;
   *)     cxxversionflag="-std=c++11 " ;;
 esac
```
Comment (amadio) [inline config/root-config.in]: If we are passing down `${cxxversion}` from CMake, then we can get rid of the `case $cxxversion` here. Otherwise, LGTM.

**Comment by dpiparo** [comment]
@phsft-bot build

**Comment by dpiparo** [comment]
@phsft-bot build

**Comment by dpiparo** [comment]
@phsft-bot build

**Pull request merged**